bundles/org.openhab.binding.anel/src/main/java/org/openhab/binding/anel/internal/discovery/AnelDiscoveryService.java

```diff
@@ -0,0 +1,221 @@
+ *
+ * @author Patrick Koenemann - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = DiscoveryService.class, immediate = true, configurationPid = "discovery.anel")
+public class AnelDiscoveryService extends AbstractDiscoveryService {
+
+    private static final String PASSWORD = "anel";
+    private static final String USER = "user7";
+    private static final int[][] DISCOVERY_PORTS = { { 750, 770 }, { 7500, 7700 }, { 7750, 7770 } };
+    private static final Set<String> BROADCAST_ADDRESSES = new TreeSet<>(NetUtil.getAllBroadcastAddresses());
+
+    private static final int DISCOVER_DEVICE_TIMEOUT_SECONDS = 2;
+
+    /** #BroadcastAddresses * DiscoverDeviceTimeout * (3 * #DiscoveryPorts) */
+    private static final int DISCOVER_TIMEOUT_SECONDS = BROADCAST_ADDRESSES.size() * DISCOVER_DEVICE_TIMEOUT_SECONDS
+            * (3 * DISCOVERY_PORTS.length);
+
+    private final Logger logger = LoggerFactory.getLogger(AnelDiscoveryService.class);
+
+    private boolean scanRunning = false;
+
+    public AnelDiscoveryService() throws IllegalArgumentException {
+        super(IAnelConstants.SUPPORTED_THING_TYPES_UIDS, DISCOVER_TIMEOUT_SECONDS);
+        logger.debug(
+                "Anel NET-PwrCtrl discovery service instantiated for broadcast addresses {} with a timeout of {} seconds.",
+                BROADCAST_ADDRESSES, DISCOVER_TIMEOUT_SECONDS);
+    }
+
+    @Override
+    protected void startScan() {
+        /*
+         * Start scan in background thread, otherwise progress is not shown in the web UI.
+         * Do not use the scheduler, otherwise further threads (for handling discovered things) are not started
+         * immediately but only after the scan is complete.
+         */
+        new Thread(this::doScan).start();
+    }
+
+    private void doScan() {
+        logger.debug("Starting scan of Anel devices via UDP broadcast messages...");
+
+        if (!scanRunning) {
+            scanRunning = true;
+
+            try {
+                for (final String broadcastAddress : BROADCAST_ADDRESSES) {
+
+                    // for each available broadcast network address try factory default ports first
+                    scan(broadcastAddress, IAnelConstants.DEFAULT_SEND_PORT, IAnelConstants.DEFAULT_RECEIVE_PORT);
+
+                    // try reasonable ports...
+                    for (int[] ports : DISCOVERY_PORTS) {
+                        int sendPort = ports[0];
+                        int receivePort = ports[1];
+
+                        // ...and continue if a device was found, maybe there is yet another device on the next port
+                        while (scan(broadcastAddress, sendPort, receivePort) || sendPort == ports[0]) {
+                            sendPort++;
+                            receivePort++;
+                        }
+                    }
+                }
+            } catch (Exception e) {
+
+                logger.warn("Unexpected exception during anel device scan", e);
```

Comment:
This will also catch `InterruptedException`, which shouldn't be unexpected. See above.